subprojects/docs/src/docs/userguide/extending-gradle/task_configuration_avoidance.adoc

```diff
@@ -57,6 +57,25 @@ If you are configuring a task by name, you will need to use the configuration av
 
 In the event you still need to get access to a Task instance, you can use link:{javadocPath}/org/gradle/api/tasks/TaskCollection.html#named-java.lang.String-[TaskCollection.named(java.lang.String)] and call link:{javadocPath}/org/gradle/api/provider/Provider.html#get--[Provider.get()].  This will cause the task to be created/configured, but everything should work as it has with the eager APIs.
 
+[[sec:how_do_i_order_tasks]]
+=== How do I order tasks?
+
+In case, when task relationships needs to be established (`dependsOn`, `finalizedBy`, `mustRunAfter`, `shouldRunAfter`), configuration avoidance can also help to avoid configuring tasks.
+Calling these methods alone _will not cause task creation_, but it's important to understand their effects, as it might seem that they do so.
```

Comment:
The latter parts of the section describes that in case of `dependsOn`, and `finalizedBy`, even if the method does not directly creates a task, it _might_ cause task instantiation.

We can debate about the ending, but I would find it important to mark at this sentence that the situation is not as clear-cut, as it _seems_.